analyses/survival-analysis/survival-analysis_histologies.Rmd

```diff
@@ -0,0 +1,234 @@
+  mutate(Tumor_Sample_Barcode = Kids_First_Biospecimen_ID_DNA) %>%
+  select(Kids_First_Biospecimen_ID_DNA, Kids_First_Biospecimen_ID_RNA, sample_id, tp53_score)
+
+# read in telomerase scores
+tel_scores <- readr::read_tsv(tel_stranded_file) %>%
+  mutate(Kids_First_Biospecimen_ID_RNA = SampleID,
+         telomerase_score = NormEXTENDScores) %>%
+  select(Kids_First_Biospecimen_ID_RNA, telomerase_score)
+
+# meta_indepine metadata, remove cell lines and samples without RNA-Seq since both classifiers' results use RNA-Seq
+# independent use primary plus rna-seq samples
+meta_indep <- tp53_scores %>%
+  left_join(tel_scores) %>%
+  left_join(histologies) %>%
+  dplyr::filter(!is.na(Kids_First_Biospecimen_ID_RNA))
+
+meta_indep <- meta_indep %>%
+  filter(Kids_First_Biospecimen_ID_RNA %in% indep_rna$Kids_First_Biospecimen_ID)
+
+```
+
+Quantiles overall
+```{r}
+meta_indep_no_hgat <- meta_indep %>%
+  filter(short_histology != "HGAT")
+
+tp53_quantiles <- quantile(meta_indep$tp53_score, probs = seq(0, 1, 0.25))
+telomerase_quantiles<- quantile(meta_indep$telomerase_score, probs = seq(0, 1, 0.25))
+
+tp53_quantiles_no_hgat <- quantile(meta_indep_no_hgat$tp53_score, probs = seq(0, 1, 0.25))
+telomerase_quantiles_no_hgat <- quantile(meta_indep_no_hgat$telomerase_score, probs = seq(0, 1, 0.25))
+```
+
+
+Create groupings for scores
+```{r}
+meta_indep <- calc_groups(meta_indep, tp53_quantiles, telomerase_quantiles)
+meta_indep_no_hgat <- calc_groups(meta_indep_no_hgat, tp53_quantiles_no_hgat, telomerase_quantiles_no_hgat)
+```
+
+Set phenotypes to loop through
+```{r}
+phenotypes <- c("tp53_0.5", "tel_0.5", "tp53_strict", "tel_strict", "pheno_0.5", "pheno_strict", "pheno_extremes")
+```
+
+Run on all PBTA and HGAT only
+## Kaplan-Meier
+
+The [Kaplan-Meier](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3059453/) 
+survival curve shows the probability of survival over time.
+
+```{r Kaplan Meier}
+
+# let's look at HGAT
+meta_all <- meta_indep
+meta_hgat <- meta_indep %>%
+  filter(short_histology == "HGAT")
+meta_no_hgat <- meta_indep_no_hgat
+
+as.data.frame(table(meta_indep$short_histology))
+
+prefix_list <- c("all-pbta","hgat","no-hgat")
+# define data frame list in the same order as prefix list 
+df_list <-list(meta_all, meta_hgat, meta_no_hgat)
+  
+for (each in phenotypes) {
+  for(i in 1:length(prefix_list)){
+    
+    prefix <- prefix_list[i]
+    df_of_interest <- df_list[[i]]
+    
+    kap_fit <- survival_analysis(df_of_interest,
+                             ind_var = each,
+                             test = "kap.meier",
+                             metadata_sample_col = "Kids_First_Biospecimen_ID_RNA"
+    )
+    
+    # median and 0.95CI survival days
+    kap_fit$model
+    
+    #`survminer` package comes with a convenient plotting function which uses `ggplot2` arguments. 
+    #It requires the original Kaplain-Meier `survfit` object, so we will extract that with `$model`.
+    surv_plot <- survminer::ggsurvplot(kap_fit$model,
```

Comment:
Something in this chunk is failing CI with the following error: `Aesthetics can not vary with a ribbon` (and a bunch of anonymous function names, which are unhelpful).

I suspect it is related to the plotting here or below, but I haven't yet had a chance to check. I wonder if it is the `palette = each` line or something else? Running this notebook in the current docker image should allow us to figure out which specific part is failing.

Reply:
I did not originally have this using 
```
palette = each
```
but rather had this as 
```
palette = "Paired", color = phenotype 
```
however, when looping, this change was required based on the documentation. This also made the labels pretty ugly, as you noted elsewhere. Open to how to improve the aesthetics for sure.

Reply:
I updated this. Will see if CI passes

Reply:
Not yet... new error is more helpful though:
```
Error in grDevices::col2rgb(colour, TRUE) : 
  invalid color name 'pheno_strict'
``` 

I think removing `color = phenotype` should fix that.

Reply:
I am starting to work on this - I will make the change and push to see whether CI passes.

Reply:
@runjin326 Great! For speed of testing, you will probably want to make sure that everything runs in the project Docker container before pushing: If the notebook works there it _should_ work in CI. Most times that isn't the case are usually related to file paths, which should not be an issue with this notebook.

Reply:
@jashapiro , thanks! Sure - then I will address all the comments, run in docker and then push to check CI. And if that does not work, we can look at the errors then and figure out a resolution. I will ping in once all the comments are addressed and pushed.

Reply:
Just to chime in here for a second – if this runs locally in Docker without issue and fails in CI, another source of failure could be the files that we use for testing. To help with debugging, you can snag the testing files: https://github.com/AlexsLemonade/OpenPBTA-analysis#working-with-the-subset-files-used-in-ci-locally

Reply:
> Just to chime in here for a second – if this runs locally in Docker without issue and fails in CI, another source of failure could be the files that we use for testing. To help with debugging, you can snag the testing files: https://github.com/AlexsLemonade/OpenPBTA-analysis#working-with-the-subset-files-used-in-ci-locally

Good point- my runs last night worked in docker so it may be the files!